lsp-mode.el

```diff
@@ -3319,32 +3319,23 @@ COMMAND-FN will be called to generate Language Server command."
                 (cons tcp-proc proc)))
    :test? (lambda () (-> command lsp-resolve-final-function lsp-server-present?))))
 
-(defun lsp-tramp-connection (command-fn)
-  "Create LSP TRAMP connection named.
-COMMAND-FN will be called to generate Language Server command."
-  (list
-   :connect (lambda (filter sentinel name)
-              (let* ((host (file-remote-p (buffer-file-name) 'host t))
-                     (port (lsp--find-available-port host (cl-incf lsp--tcp-port)))
-                     (command (funcall command-fn name port))
-                     (proc (progn
-                             (lsp--info "Starting %s" command)
-                             (async-shell-command command
-                                                  (generate-new-buffer-name (format "%s::stdout" name))
-                                                  (generate-new-buffer-name (format "%s::stderr" name)))))
-                     (tcp-proc (lsp--open-network-stream host port (concat name "::tramp::tcp"))))
-
-                (set-process-sentinel proc sentinel)
-                (set-process-query-on-exit-flag proc nil)
-                (set-process-query-on-exit-flag tcp-proc nil)
-                (set-process-filter tcp-proc filter)
-                (cons tcp-proc proc)))
-   :test? (-const t)))
-
-(defun lsp-make-nc-tramp-command (program)
-  "Return a function which will convert PROGRAM from STDIO command to TCP one."
-  (lambda (_name port)
-    (format "nc -l -p %s -c '%s'" port program)))
+(defun lsp-tramp-connection (local-command)
+  "Create LSP stdio connection named name.
+COMMAND is either list of strings, string or function which
+returns the command to execute."
+  (list :connect (lambda (filter sentinel name)
+                   (let* ((final-command (lsp-resolve-final-function local-command))
+                          ;; wrap with stty to disable converting \r to \n and line buffering
+                          (wrapped-command (append '("stty" "-icrnl" "extproc" ";") final-command))
```

Comment:
My stty (8.22) is probably too old, is there any other way to workaround this?

```
Warning (lsp-mode): Failed to parse the following chunk:
’’’
stty: invalid argument ‘extproc’
Try 'stty --help' for more information.
Content-Length: 152

{"jsonrpc":"2.0","method":"$cquery/progress","params":{"indexRequestCount":0,"doIdMapCount":0,"onIdMappedCount":0,"onIndexedCount":0,"activeThreads":0}}
’’’
with message (error No Content-Length header)
```

Reply:
This is indeed a recent stty feature, which is unfortunate. I changed the fix for this back to adding whitespace to the payload, which works independently of the stty version.

Thanks for testing early and catching this!